test/message/console_low_stack_space.js

```diff
@@ -6,17 +6,26 @@ global.console = {};
 
 require('../common');
 
+// This test checks that, if Node cannot put together the `console` object
+// because it is low on stack space while doing so, it can succeed later
+// once the stack has unwound a little, and `console` is in a usable state then.
+
+let compiledConsole;
+
 function a() {
   try {
     return a();
   } catch (e) {
-    const console = consoleDescriptor.get();
-    if (console.log) {
-      console.log('Hello, World!');
+    compiledConsole = consoleDescriptor.get();
+    if (compiledConsole.log) {
+      // Using `console.log` itself might not succeed yet, but the code for it
+      // has been compiled.
     } else {
```

Comment:
Can we remove this branch?

```js
if (!compiledConsole.log) {
  throw e;
}
```

If you think it's clearer as is, no problem, ignore the comment.